### PR TITLE
findAllReferences: Type parameter is not globally visible

### PR DESCRIPTION
**Pull request by andy-ms**
Normally a symbol with a parent is exposed by that parent, but type parameters can only be referenced within their container.

### CODE REVIEW

**Comment by mhegazy** [comment]
that is not the case of reopened declarations, e.g.:
```ts
interface I <T> { a: T }
interface I <T> { b: T }
```

**Comment by andy-ms** [comment]
Added a test that that scenario still works.

File: src/services/findAllReferences.ts [hunk 1]
```diff
@@ -648,9 +648,11 @@ namespace ts.FindAllReferences.Core {
         }
 
         // If the symbol has a parent, it's globally visible.
+        // (Private properties handled above.)
```
Comment (aozgaa) [inline src/services/findAllReferences.ts]: Why is this comment helpful here?

Comment (andy-ms) [inline src/services/findAllReferences.ts]: I just figured if someone read "If the symbol has a parent, it's globally visible.", they would try thinking of an exception -- want me to remove? Or I could just reword it to "If a non-private symbol has a parent, it's globally visible, unless it's a type parameter."
